database/database.go

```diff
@@ -0,0 +1,128 @@
+package database
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/cabify/couchdb-admin/array_utils"
+	"github.com/cabify/couchdb-admin/cluster"
+	"github.com/cabify/couchdb-admin/http_utils"
+)
+
+type Database struct {
+	name   string
+	config Config
+}
+
+type Config struct {
+	Id        string              `json:"_id"`
+	Rev       string              `json:"_rev"`
+	Shards    []int               `json:"shard_suffix"`
+	Changelog [][]string          `json:"changelog"`
+	ByNode    map[string][]string `json:"by_node"`
+	ByRange   map[string][]string `json:"by_range"`
+}
+
+func LoadDB(name string, ahr *http_utils.AuthenticatedHttpRequester) *Database {
+	db := &Database{
+		name: name,
+	}
+	db.refreshDbConfig(ahr)
+	return db
+}
+
+func CreateDatabase(name string, replicas, shards int, ahr *http_utils.AuthenticatedHttpRequester) *Database {
+	req, err := http.NewRequest("PUT", fmt.Sprintf("http://%s:5984/%s?n=%d&q=%d", ahr.GetServer(), name, replicas, shards), nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ahr.RunRequest(req, nil)
+
+	return LoadDB(name, ahr)
+}
+
+func (db *Database) refreshDbConfig(ahr *http_utils.AuthenticatedHttpRequester) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("http://%s:5986/_dbs/%s", ahr.GetServer(), db.name), nil)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	ahr.RunRequest(req, &db.config)
+}
+
+func (db *Database) Replicate(shard, replica string, ahr *http_utils.AuthenticatedHttpRequester) error {
+	replica_node := fmt.Sprintf("couchdb@%s", replica)
+
+	if array_utils.Contains(db.config.ByNode[replica_node], shard) {
```

Comment:
Same here. Go encourages CamelCased variables. `sliceUtils` should be more correct